workspace/src/main/kotlin/com/cosmotech/workspace/service/WorkspaceServiceImpl.kt

```diff
@@ -215,6 +215,10 @@ internal class WorkspaceServiceImpl(
       throw IllegalArgumentException("Invalid destination: '$destination'. '..' is not allowed")
     }
     val workspace = getVerifiedWorkspace(organizationId, workspaceId, PERMISSION_WRITE)
+    if (file?.filename?.contains("..") == true || file?.filename?.contains("/") == true) {
```

Comment:
Would it be useful to check for "\" too ?

Reply:
The quote is a possible character for a filename (even though not very useful), and since it cannot be abused to write to a different location I think we should keep it